### ISSUE TITLE
IDE0161 should be self-sufficient and documented

### ISSUE THREAD

**Issue by BrightSoul** [dotnet/roslyn#60546]
I've tried to use rule IDE0161 to enforce file-scoped namespaces in my .NET 6 project.
This is the content of my `.editorconfig`:
```
[*.cs]
dotnet_diagnostic.IDE0161.severity = warning
```
However, no warning is emitted when I'm building the project containing .cs files with block-style namespaces. The `.editorconfig` also needs this other line:
```
csharp_style_namespace_declarations = file_scoped
```
Why is this the case? Shouldn't the `dotnet_diagnostic.IDE0161.severity = warning` rule be sufficient to emit warnings.

Also, I can't find any documentation on the Microsoft Docs for this rule.
<img width="479" alt="IDE0161" src="https://user-images.githubusercontent.com/1888641/161425097-40b180dc-711e-416e-950f-56c92dab45b7.png">

Moreno

**Comment by Youssef1313** [comment]
The default is to prefer block scope

https://github.com/dotnet/roslyn/blob/9077a36fb47418cf1a9d96f5ad0499740e36df9b/src/Workspaces/SharedUtilitiesAndExtensions/Compiler/CSharp/CodeStyle/CSharpCodeStyleOptions.cs#L337-L340

So you have to change it. 

Documentation is unfortunately currently missing for this analyzer and some few more. FYI @gewarren

**Comment by BrightSoul** [comment]
Hey @Youssef1313, thanks for your reply!

> The default is to prefer block scope [...] So you have to change it.

That seems redundant since IDE0161 is specific about using file-scoped namespaces.
https://github.com/dotnet/roslyn/blob/33ff264c786281d9c2c1688d47729211713c1e43/src/Analyzers/Core/Analyzers/IDEDiagnosticIds.cs#L164

Why should I reiterate the very same preference by using another rule?
If, instead, I preferred block-style namespaces, I would have used IDE0160.
https://github.com/dotnet/roslyn/blob/33ff264c786281d9c2c1688d47729211713c1e43/src/Analyzers/Core/Analyzers/IDEDiagnosticIds.cs#L163

Thanks!
Moreno

**Comment by neon-sunset** [comment]
> The default is to prefer block scope

Why is this the case for new projects/default editorconfig?

**Comment by CyrusNajmabadi** [comment]
> Why is this the case for new projects/default editorconfig?

Not sure what you're asking.  But the reason is so that the code we generate is consistent with 99.99% of these code out there. If users want to use the new style, they just enable this option and are done.

**Comment by nicolaslattuada** [comment]
Hello 
I run into a similar issue, I use ruleset format and I want to enforce namespace declaration format with block_scoped style. Unfortunately this is not possible, even if I set IDE0161 to Error, no errors are produced.
On the other hand if I set IDE0160 errors does get produced, but it's not my format of choice :(

**Comment by Youssef1313** [comment]
This is now documented and should be closed. https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/style-rules/ide0160-ide0161

**Comment by Youssef1313** [comment]
@nicolaslattuada The same code-style option is shared between both IDE0160 and IDE0161

**Comment by nicolaslattuada** [comment]
thanks @Youssef1313 , my point is that it should not.

The use case for enforcing block-style is covered by setting IDE0160  to error.

The use case for enforcing file-scoped **should be** covered by setting IDE0161  to error.

The current status is that in order to enforce file-scoped, you need to both set IDE0161 to error **and change code style option**

IMO to be consistent to  code style option of IDE0161 should be opposite.

**Comment by squishleheimer** [comment]
Did anyone find a workaround for having to have the `csharp_style_namespace_declarations = file_scoped` in the .editorconfig in addtion to the severiuty setting?

It's fairly naff having a rule that also requires the additional directive in another file.

**Comment by CyrusNajmabadi** [comment]
No. This is by design.  The editorconfig option states what behavior is preferred. The diagnostic is reported based on that.
